### ISSUE TITLE
Guard clause in a middle of function?

### ISSUE THREAD

**Issue by sandric** [rubocop-hq/ruby-style-guide#813]
Recently I stubbed on issue that I should consider guard clause refactoring this:

```ruby
def execute
  some_code_1_line
  some_code_2_line

  if some_condition
    some_code_inside_if
  end
end
```
to this

```ruby
def execute
  some_code_1_line
  some_code_2_line

  return if some_condition
  
  some_code_inside_if
end
```

The thing is that mine rubocop config does have `guard_clause` rule, but it does not show any refactoring warning for this line. So my questions are is this guard-clause refactoring, is rubocop cop should be changed to consider all returns not only at beginning of function, and whats `ruby-style-guide` consider using in such case. Thanks.

**Comment by pirj** [comment]
I guess it should become:

```ruby
  return unless some_condition
  
  some_code_inside_if
```
`unless`, not `if`.

Maybe it's a deficiency in the cop, I suggest opening a ticket for RuboCop.

Otherwise closing as stale.
